### ISSUE TITLE
Window peek on hovering over preview

### ISSUE THREAD

**Issue by robrobinbin** [jderose9/dash-to-panel#97]
On Windows, hovering over the preview of a window brings that window to the front and fades all others. This would be a useful feature for this extension. Similar behaviour was implemented in [this cinnamon applet](https://github.com/linuxmint/cinnamon-spices-applets/tree/master/IcingTaskManager@json)

**Comment by jderose9** [comment]
Great idea, thanks!

**Comment by dziku1337** [comment]
Hi, I'm kinda looking into this feature. In windows you enter the "peek mode" if you stop (Or your speed is very low, dunno) your pointer while hovering over a thumbnail preview. You exit the "peek mode" and go back to normal when you move the pointer out of thumbnail preview list.  

Do you know if there's a neat way to check mouse pointer's speed? Well, you can always keep track of pointer's position and determine the speed yourself but that is a nuisance. I'm totally new to the gnome extensions etc. so I don't know much but still would like to help.

**Comment by jderose9** [comment]
@dziku1337 I'm not sure if there is a way to do that. What I would try is:

1. Handle the enter-event notification on the preview thumbnail and set a timeout on Mainloop that will show the peek mode after X timeout (say 100 or 200 ms), and save it's ID in a variable.
2. Handle the motion-event to cancel the timeout and re-add it. So basically every time mouse movement is detected, your timer will be reset.
3. Handle the leave-event to cancel the timeout.

So basically peek mode would trigger if they stay within the thumbnail for X ms without mouse movement being detected.

The motion-event likely only fires after some X amount of movement so you don't get flooded with events or detect motion when they only move a pixel or two, but I haven't tried that to confirm.

**Comment by dziku1337** [comment]
@jderose9 Is there some way to contact you outside of github? I've got some questions concerning your code, but issues aren't really meant for chatting.

**Comment by jderose9** [comment]
@dziku1337 You can email me at coder@derose.io. But, if you've got a question that might benefit others who want to work on the extension, you could post it in 58 or a new issue. Thanks!

**Comment by dziku1337** [comment]
On windows 7 and 8 when you enter the peek mode, all the windows on all the screens hide with exception of the one on which thumbnail you are currently hovering the pointer over. There are no workspaces on win 7 and 8.

But what would happen when you're working with workspaces? Let's say that I've got two different workspaces on two screens, Does peek mode hide all windows in all workspaces or does it only hide the windows in the current workspace?

**Comment by jderose9** [comment]
@dziku1337 Assuming that the goal of this feature is to allow quick visual identification of the window and its location that will be focused once the thumbnail is clicked, I would say that all currently visible windows should be hidden regardless of whether you have a different workspace on each monitor.

**Comment by dziku1337** [comment]
I have submitted a pull request for this feature.

**Comment by jderose9** [comment]
@dziku1337 Thanks so much for this! It is awesome.
